models/BaseEntity.cfc

```diff
@@ -1322,6 +1366,23 @@ component accessors="true" {
 	 * @return  quick.models.BaseEntity
 	 */
 	public any function save() {
+		if ( variables._hasParentEntity ) {
+			if ( isLoaded() ) {
+				var parent = variables._wirebox
+					.getInstance( variables._parentEntity.meta.fullName )
+					.set_LoadChildren( false )
+					.findOrFail( keyValues() );
+			} else {
+				var parent = variables._wirebox.getInstance( variables._parentEntity.meta.fullName );
+			}
+
+			parent.fill( getMemento(), true ).save();
```

Comment:
I wonder if we could use [`updateOrCreate`](https://quick.ortusbooks.com/guide-1/getting-started/creating-new-entities#updateorcreate) here?

Reply:
Let me give that a try.

Reply:
I just tried this and it's a no-go because `updateOrCreate` treats an empty key value ( from the getMemento() ) as the authoritative value when it fills and then the entity returns an empty string for the key.  I also can't  use it without some parent introspection beforehand and conditional struct assembly.  

I think both `updateOrCreate` and `firstOrNew` could be modified to deal with the issue, but my concern would be the possibility of bugs when dealing with different key types - especially string key types like UUID.